kv/kvclient/kvcoord: TestGetFirstRangeDescriptor failed

kv/kvclient/kvcoord.TestGetFirstRangeDescriptor [failed](https://tanzanite.cluster.engflow.com/invocations/default/c86b4362-1a3c-4077-84bc-e3559b00645a?testReportRun=2&testReportShard=13&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZjbGllbnQva3Zjb29yZDprdmNvb3JkX3Rlc3Q=) on release-24.3.15-rc @ [b33c70a891bb565033304c0411af9875506051f3](https://github.com/cockroachdb/cockroach/commits/b33c70a891bb565033304c0411af9875506051f3):


```
=== RUN   TestGetFirstRangeDescriptor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestGetFirstRangeDescriptor1164274586
    test_log_scope.go:76: use -show-logs to present logs inline
    dist_sender_test.go:2100: could not get range descriptor after 25 cycles
    dist_sender_test.go:2112: -- test log scope end --
test logs left over in: outputs.zip/logTestGetFirstRangeDescriptor1164274586
--- FAIL: TestGetFirstRangeDescriptor (0.78s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=13</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGetFirstRangeDescriptor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-51885

--- a/pkg/kv/kvclient/kvcoord/dist_sender_test.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_test.go
@@ -2072,6 +2072,11 @@ func TestGetFirstRangeDescriptor(t *testing.T) {
 		node.Gossip.EnableSimulationCycler(false)
 	}
 	n.Start()
+	// Make sure the first two nodes are connected via gossip.
+	n.SimulateNetwork(func(_ int, _ *simulation.Network) bool { return false })
+	<-n.Nodes[0].Gossip.Connected
+	<-n.Nodes[1].Gossip.Connected
+
 	ds := NewDistSender(DistSenderConfig{
 		AmbientCtx:         log.MakeTestingAmbientContext(stopper.Tracer()),
 		NodeDescs:          n.Nodes[0].Gossip,

